### DISCUSSION TITLE
Private certificate authority

### DISCUSSION THREAD

**Discussion by kunert7062** [MichaIng/DietPi#6290] [Q&A]
Hey all. Letsencrypt is awesome, but doesn't work on private networks. I've been running bitwarden server for a couple years with no problem, until it became vaultwarden. Now my app no longer works because it now requires https.

How about a quick to install, easy to use personal authority for those locked away from the net, but still needing encryption services? Or those who wish to test, but not directly online.

it seems to me forcing folks to put personal, hobbyist sites online just so https will work locally is a bit silly, opening folks up to threats they likely don't comprehend. I have two encrypted sites running now on my home net. If I need remote access, I prefer to tunnel in through SSH than have to worry about online attacks.

What do you say, Micha? Something to create and sign the cert and serve validation requests? XCA seems fairly decent, but if you gave it the Dietpi treatment, where store locations are known, and a copy can easily be created to share with clients, that'd be grand.

But I won't hold my breath.

**Comment by Joulinar** [comment]
Vaulwarden is working perfectly with the self signed certificates we create. We created some hints on our online docs how to extract the cert and import it into various OS to enable Bitwarden clients to work still.

**Comment by MichaIng** [comment]
Indeed, for vaultwarden a self-signed certificate is created automatically on package install, or was there an issue with this in your case?

I also had the idea to allow creating self-signed certificates as alternative to Let's Encrypt via dedicated script, e.g. renaming `dietpi-letsencrypt` to `dietpi-https` or something like that where one can the choose how it is created. Would contain [acme.sh](https://github.com/acmesh-official/acme.sh), a lightweight alternative to Certbot which is use on my servers already.

**Reply by Joulinar** [reply]
but i don't think this will change anything on Vaultwarden because self-signed certificates are not accepted by most Bitwarden apps if I'm not mistaken.

**Reply by MichaIng** [reply]
If I remember right, the Bitwarden (Windows) desktop app and of course the web vault did accept it. I think I tested a mobile client as well. But for one client (likely several) one needed to import the certificate into the OSes trusted store. This part indeed is way to difficult IMO, but nothing we could do anything about.
